src/miradorDownloadPlugin.js

```diff
@@ -1,140 +1,67 @@
-    if (!(
-      manifest
-      && manifest.getSequences()
-      && manifest.getSequences()[0]
-      && manifest.getSequences()[0].getRenderings()
-    )) return [];
+class MiradorDownload extends Component {
+  openDialogAndCloseMenu() {
+    const { handleClose, openDownloadDialog } = this.props;
 
-    return manifest.getSequences()[0].getRenderings();
+    openDownloadDialog();
+    handleClose();
   }
 
   render() {
-    const {
-      canvases, canvasLabel, classes, viewType, windowId,
-    } = this.props;
-    const { modalDisplayed } = this.state;
-
     return (
       <React.Fragment>
-        <MenuItem onClick={this.handleClick}>
+        <MenuItem onClick={() => this.openDialogAndCloseMenu()}>
           <DownloadIcon />
           <ListItemText inset primaryTypographyProps={{ variant: 'body1' }}>
             Download
           </ListItemText>
         </MenuItem>
-        <Dialog
-          disableEnforceFocus
-          onClose={this.handleDialogClose}
-          open={modalDisplayed}
-          scroll="paper"
-          fullWidth
-          maxWidth="xs"
-        >
-          <DialogTitle disableTypography className={classes.h2}>
-            <Typography variant="h2">Download</Typography>
-          </DialogTitle>
-          <DialogContent>
-            {canvases.map(canvas => (
-              <CanvasDownloadLinks
-                canvas={canvas}
-                canvasLabel={canvasLabel(canvas.index)}
-                classes={classes}
-                key={canvas.id}
-                viewType={viewType}
-                windowId={windowId}
-              />
-            ))}
-            {this.renderings().length > 0
-              && <ManifestDownloadLinks classes={classes} renderings={this.renderings()} />
-            }
-          </DialogContent>
-          <DialogActions>
-            <Button onClick={this.handleDialogClose} color="primary">
-              Close
-            </Button>
-          </DialogActions>
-        </Dialog>
       </React.Fragment>
     );
   }
 }
 
 MiradorDownload.propTypes = {
-  canvasLabel: PropTypes.func.isRequired,
-  canvases: PropTypes.arrayOf(
-    PropTypes.shape({ id: PropTypes.string, index: PropTypes.number }),
-  ).isRequired,
-  classes: PropTypes.shape({
-    h2: PropTypes.string,
-    h3: PropTypes.string,
-  }).isRequired,
-  manifest: PropTypes.shape({
-    getSequences: PropTypes.func.isRequired,
-  }).isRequired,
-  viewType: PropTypes.string.isRequired,
-  windowId: PropTypes.string.isRequired,
+  handleClose: PropTypes.func.isRequired,
+  openDownloadDialog: PropTypes.func.isRequired,
 };
 
-const styles = () => ({
-  h2: {
-    paddingBottom: 0,
-  },
-  h3: {
-    marginTop: '20px',
-  },
-});
-
 export default {
   target: 'WindowTopMenu',
   mode: 'add',
-  component: withStyles(styles)(MiradorDownload),
-  mapStateToProps,
+  component: MiradorDownload,
+  mapDispatchToProps,
+  reducers: {
+    windowDialogs: downloadDialogReducer,
```

Comment:
One thing that we'll want to keep an eye on is how Mirador handles this reducer collision.  Luckily the reducers in the download plugin and the share plugin are identical, so it shouldn't be an issue (but that does indicate that the actions/reducer may be a useful feature to have in Mirador core)